R/data.table.R

```diff
@@ -253,7 +253,7 @@ replace_dot_alias = function(e) {
   }
   if (!missing(j)) {
     jsub = replace_dot_alias(jsub)
-    root = if (is.call(jsub)) as.character(jsub[[1L]])[1L] else ""
```

Comment:
This is where 1128 fails. Before, root evaluated to `(` but now it evaluates to an empty string. 

``` r
## check for #88
DT = data.table(a=0:2,b=3:5,key="a")
test(1128, DT[, (function(){b})()], DT[, b])
```

If we include the original logic - if it is a call (but `jsub[[1L]]` is not a name!) - we can get all tests to work again. I apologize for formatting - it might be a bit much for a one-liner now...

``` r
root_name = function(jsub) if (is.call(jsub)) if (is.name(j1 <- jsub[[1L]])) as.character(j1) else as.character(jsub[[1L]])[1L] else ""
```

Reply:
No need to be one line :)

I'm suspecting now we might be breaking something with the current approach, it's strange to need to handle `(` and only `(` this way...